src/DataSet.php

```diff
@@ -204,18 +204,19 @@ public function build()
 
             case "query":
             case "model":
-                $this->total_rows = $this->query->count();
```

Comment:
@zofe this query is duplicated. we can get total rows value from paginator instance.